File: backend/app/schemas/staff.py
from __future__ import annotations
from datetime import datetime
from pydantic import BaseModel, EmailStr
from app.enums import Role


class StaffCreate(BaseModel):
    """Create new staff member (Manager or Sales)"""
    email: EmailStr
    password: str
    role: Role

    class Config:
        use_enum_values = False


class StaffOut(BaseModel):
    """Staff member info"""
    id: int
    user_id: int
    supplier_id: int
    role: Role
    invited_by: int
    created_at: datetime
    user: 'UserBasic | None' = None
    inviter: 'UserBasic | None' = None

    class Config:
        from_attributes = True


class StaffUpdate(BaseModel):
    """Update staff role"""
    role: Role


# Resolve forward references
from app.schemas.user import UserBasic
StaffOut.model_rebuild()

File: frontend/types/staff.ts
import { Role } from '@/enums';

export interface Staff {
  id: number;
  user_id: number;
  supplier_id: number;
  role: Role;
  invited_by: number;
  created_at: string;
  user: {
    id: number;
    email: string;
    name?: string;
  };
  inviter: {
    id: number;
    email: string;
    name?: string;
  };
}

export interface StaffCreate {
  email: string;
  password: string;
  role: Role.SUPPLIER_MANAGER | Role.SUPPLIER_SALES;
}

export interface StaffUpdate {
  role: Role.SUPPLIER_MANAGER | Role.SUPPLIER_SALES;
}
